Add contributing guidelines and switch to DCO

jdemel opened the pull request:
It is recommended good practice to have a CONTRIBUTING.md file. I used the GNU Radio file as a template and adopted and simplified it a bit. Since GNU Radio switch to DCO, I assume it is a welcome improvement for VOLK as well.

Still, be aware that this is quite a big step for VOLK besides the fact this is just another PR.

I added a DCO.txt as well. I suggest we move from a CLA based contribution system to a DCO based one. I hope this decreases the entry hurdle to new contributors.

Just like GNU Radio I'd like to use [DCObot](https://probot.github.io/apps/dco/) for VOLK as well.

jdemel committed:
contributing: Add CONTRIBUTING.md and DCO.txt

It is recommended good practice to have a CONTRIBUTING.md file. I used
the GNU Radio file as a template and adopted and simplified it a bit.
Since GNU Radio switch to DCO, I assume it is a welcome improvement for
VOLK as well.

I added a DCO.txt as well. I suggest we move from a CLA based
contribution system to a DCO based one. I hope this decreases the entry
hurdle to new contributors.

Just like GNU Radio I'd like to use [DCObot][https://probot.github.io/apps/dco/]
for VOLK as well.

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

rear1019 commented on CONTRIBUTING.md:
This only applies to the `gnuradio` repository.

jdemel replied:
Yeah. I'll remove it. Thanks.

jdemel committed:
contributing: Remove gnuradio specific section

VOLK does not have any GitHub specific teams. We just host our code on
github.com.

michaelld reviewed (approved):
LGTM ; makes sense & I think it'll simplify contributions!
